### PR TITLE
Add support for `bevy_xpbd`

### PR DESCRIPTION
**Pull request by Jondolf**
Depends on #21; merge that first.

This PR adds support for `bevy_xpbd` as an alternative physics backend.

## Changes

- Added `xpbd_collisions` feature
- Added `rapier` and `xpbd` sub-modules for `collisions`
- Added `xpbd_collision` example; a clone of `rapier_example`, but modified for `bevy_xpbd`
- Updated `README.md` and `src/lib.rs` to have separate sections for `bevy_rapier` and `bevy_xpbd`. If you'd like it to be changed more, let me know :)

### CODE REVIEW

**Comment by ManevilleF** [comment]
Thanks for your contribution, should I merge #21 first ? And then you'll rebase?

File: src/systems/collisions/xpbd.rs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    clippy::type_complexity,
+    clippy::option_if_let_else,
+    clippy::suboptimal_flops
+)]
+use crate::components::{cloth::Cloth, collider::ClothCollider};
+use bevy::{log, prelude::*, render::primitives::Aabb};
+use bevy_xpbd_3d::prelude::*;
+
+fn get_collider(aabb: &Aabb, collider: &ClothCollider) -> Collider {
+    let extents = aabb.half_extents * 2.0 + collider.offset;
```
Comment (ManevilleF) [inline src/systems/collisions/xpbd.rs:12]: Why is the * 2.0 needed? We don't do that in rapier

Comment (Jondolf) [reply src/systems/collisions/xpbd.rs:12]: `Collider::cuboid` currently takes full extents to match Bevy's shape APIs better

**Comment by Jondolf** [comment]
Yeah, you can merge that one first. I'll fix the format warning here and add these changes to the changelog as well, and then rebase

**Review by ManevilleF — APPROVED** [review]

**Comment by ManevilleF** [comment]
Btw to synchronize the README and lib.rs I use cargo-sync-readme

**Pull request merged**